catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/TemplateTransformer.java

```diff
@@ -99,11 +128,30 @@ public Metacard toQueryTemplateMetacard(Map<String, Object> formTemplate) {
       if (querySettings != null) {
         metacard.setQuerySettings(querySettings);
       }
+
+      Result result = metacardAlreadyExists(id, filterBuilder, catalogFramework);
```

Comment:
✏️ `fetchExistingMetacard` or something might be a better name for this method since the current name seems to imply that it's a boolean check of some kind

Reply:
What about `getMetacardIfExistsOrNull` like in your other method?